synapse/events/spamcheck.py

```diff
@@ -214,3 +216,47 @@ async def check_registration_for_spam(
                     return behaviour
 
         return RegistrationBehaviour.ALLOW
+
+    async def check_media_file_for_spam(
+        self, file_wrapper: ReadableFileWrapper, file_info: FileInfo
+    ) -> bool:
+        """Checks if a piece of newly uploaded media should be blocked.
+
+        This will be called for local uploads, downloads of remote media, each
+        thumbnail generated for those, and web pages/images used for URL
+        previews.
+
+        Note that care should be taken to not do blocking IO operations in the
+        main thread. For example, to get the contents of a file a module
+        should do::
+
+            async def check_media_file_for_spam(
+                self, file: ReadableFileWrapper, file_info: FileInfo
+            ) -> bool:
+                buffer = BytesIO()
+                await file.write_to_io(buffer)
+
+                if bytes.getvalue() == b"Hello World":
+                    return True
+
+                return False
+
+
+        Args:
+            file: An object that allows reading the contents of the media.
+            file_info: Metadata about the file.
+
+        Returns:
+            True if the media should be blocked or False if it should be
+            allowed.
+        """
+
+        for spam_checker in self.spam_checkers:
+            # For backwards compatibility, only run if the method exists on the
+            # spam checker
+            checker = getattr(spam_checker, "check_media_file_for_spam", None)
+            spam = await maybe_awaitable(checker(file_wrapper, file_info))
```

Comment:
I think this code might not really work properly with multiple spam checkers?

Reply:
Why? Though I have forgotten to add an if check there

Reply:
Oh, this should be OK. I completely misread how to use the `ReadableFileWrapper` API. It is a one-shot to read the file in chunks. I thought it returned only a single chunk and you had to keep calling it.

My only comment would be that this could be inefficient with multiple spam checkers, but that's not a huge deal.